backend/infrahub/core/migrations/schema/node_attribute_add.py

```diff
@@ -29,3 +41,41 @@ def __init__(
 class NodeAttributeAddMigration(AttributeSchemaMigration):
     name: str = "node.attribute.add"
     queries: Sequence[type[AttributeMigrationQuery]] = [NodeAttributeAddMigrationQuery01]  # type: ignore[assignment]
+
+    async def execute_post_queries(
+        self,
+        db: InfrahubDatabase,
+        result: MigrationResult,
+        branch: Branch,
+        at: Timestamp,  # noqa: ARG002
+    ) -> MigrationResult:
+        if self.new_attribute_schema.kind != "NumberPool":
+            return result
+
+        number_pool: CoreNumberPool = await Node.fetch_or_create_number_pool(  # type: ignore[assignment]
+            db=db, branch=branch, schema_node=self.new_schema, schema_attribute=self.new_attribute_schema
+        )
+
+        await update_branch_registry(db=db, branch=branch)
```

Comment:
In the `refresh_branches` function we run this code within a local schema lock. We should use the same lock here when we update the registry schema.

I'm not sure about the timing if this function above would run prior to `refresh_branches`. If it does we'd avoid the extra load as the hash would already be updated once the lock is released. But not having the same lock here seems problematic.

Reply:
Ok but does this also produce the same deadlock?

```python
async with lock.registry.local_schema_lock():
    await update_branch_registry(db=db, branch=branch)
```

Reply:
yes it's the same, the issue is only present during the functional test